pkg/policy/groups/helpers.go

```diff
@@ -44,7 +44,9 @@ func getDerivativeName(cnp *cilium_v2.CiliumNetworkPolicy) string {
 }
 
 // createDerivativeCNP will return a new CNP based on the given rule.
-func createDerivativeCNP(ctx context.Context, cnp *cilium_v2.CiliumNetworkPolicy) (*cilium_v2.CiliumNetworkPolicy, error) {
+// ccnpDerived argument indicates if the provided CNP is derived from CCNP or not.
+// This is to make sure we call `Parse()` for the right construct so it don't fail.
+func createDerivativeCNP(ctx context.Context, cnp *cilium_v2.CiliumNetworkPolicy, ccnpDerived bool) (*cilium_v2.CiliumNetworkPolicy, error) {
```

Comment:
It's probably better to create another function that `createDerivativeCCNP`. My main concern is that we are mixing up the 2 types (CCNP and CNP) and we might create bugs because of that.

Discussed offline.